### DISCUSSION TITLE
API definition

### DISCUSSION THREAD

**Discussion by janveijck** [soundshed/soundshed-app#71] [Q&A]
Hi,
Am trying to fill my M-Vave Chocolate's slots (128) with Spark presets by storing them on an ESP32. Now I found here the API call to get for instance the top-100 most popular presets (API.positivegrid.com/V2/preset). Thank you very much for this. 
However, this shows me the effect chain with the different components. But I don't get back the individual effect parameters (like amp gain=5). Is there an additional call to the API that needs to be made? Would you have the API definition perhaps you'd be willing to share?
Many thanks from a fellow enthusiast. 

Jan

**Comment by webprofusion-chrisc** [comment]
Hi Jan,

I think it's `/preset/{id}`

See also https://github.com/soundshed/soundshed-app/blob/6dc155feb78d7084b2bcbb7dd221244f3347afa7/src/spork/src/devices/spark/sparkAPI.ts#L143

There is also some useful stuff on how their "dsp id" maps to specific fx and what parameters they have:
https://github.com/soundshed/soundshed-app/blob/main/src/spork/src/devices/spark/sparkFxCatalog.ts

Note that preset upload to the spark takes a second or so, so for me for live use the preset upload+switch latency is too low to be useable.

Good luck!
